lib/spack/spack/test/config.py

```diff
@@ -1431,3 +1435,54 @@ def test_config_file_read_invalid_yaml(tmpdir, mutable_empty_config):
 
     with pytest.raises(spack.config.ConfigFileError, match="parsing yaml"):
         spack.config.read_config_file(filename)
+
+
+def test_environment_created_in_users_location(mutable_config, tmpdir):
+    """Test that an environment is created in a location based on the config"""
+    spack.config.set("config:environments_root", str(tmpdir.join("envs")))
+    env_dir = spack.config.get("config:environments_root")
+
+    assert tmpdir.strpath in env_dir
+    assert not os.path.isdir(env_dir)
+
+    os.makedirs(env_dir)
+    env("create", "test")
+    out = env("list")
+
+    assert "test" in out
+    assert env_dir in ev.root("test")
+    assert os.path.isdir(os.path.join(env_dir, "test"))
+
+
+def test_environment_errors_if_root_missing(mutable_config, tmpdir):
+    spack.config.set("config:environments_root", str(tmpdir.join("envs")))
+    env_dir = spack.config.get("config:environments_root")
+
+    assert tmpdir.strpath in env_dir
+    assert not os.path.isdir(env_dir)
+
+    with pytest.raises(ev.SpackEnvironmentError, match=env_dir):
+        env("create", "test")
+
+
+def test_environment_cant_modify_environments_root(tmpdir):
+    filename = tmpdir.join("spack.yaml")
+    with open(filename, "w") as f:
+        f.write(
+            """\
+spack:
+  config:
+    environments_root: /a/black/hole
+  view: false
+  specs: []
+"""
+        )
+    with tmpdir.as_cwd():
+        with pytest.raises(ev.SpackEnvironmentError):
+            # Why does this raise?
+            e = ev.Environment(tmpdir.strpath)
+            ev.activate(e)
+        with pytest.raises(ev.SpackEnvironmentError):
+            # When this does not?
+            env("create", "-d", tmpdir.strpath)
+            env("activate", "-d", tmpdir.strpath)
```

Comment:
😵‍💫

Reply:
So after digging into this it looks like the `SpackCommand('env')` is really doing shell calls so my test is not actually getting executed in this shell for pytest to catch.